diff --git a/xen/Rules.mk b/xen/Rules.mk
index 9d0e6ac..5b5fa2e 100644
--- a/xen/Rules.mk
+++ b/xen/Rules.mk
@@ -223,11 +223,11 @@ SPECIAL_DATA_SECTIONS := rodata $(foreach a,1 2 4 8 16, \
 			 $(foreach r,rel rel.ro,data.$(r).local)
 
 $(filter %.init.o,$(obj-y) $(obj-bin-y) $(extra-y)): %.init.o: %.o Makefile
-	$(OBJDUMP) -h $< | sed -n '/[0-9]/{s,00*,0,g;p;}' | while read idx name sz rest; do \
+	$(OBJDUMP) -h $< | while read idx name sz rest; do \
 		case "$$name" in \
 		.*.local) ;; \
 		.text|.text.*|.data|.data.*|.bss) \
-			test $$sz != 0 || continue; \
+			test $$(echo $$sz | sed 's,00*,0,') != 0 || continue; \
 			echo "Error: size of $<:$$name is 0x$$sz" >&2; \
 			exit $$(expr $$idx + 1);; \
 		esac; \
